libs/ngxtension/computed-async/src/computed-async.ts

```diff
@@ -72,46 +96,157 @@ interface ComputedAsyncOptions<T> extends CreateComputedOptions<T> {
+): Signal<T>;
+
+export function computedAsync<T>(
+	computation: (
+		previousValue?: T | undefined,
+	) => Promise<T> | Observable<T> | T | undefined,
+	options?: any,
+): Signal<T> {
 	return assertInjector(computedAsync, options?.injector, () => {
 		const destroyRef = inject(DestroyRef);
 
 		// source$ is a Subject that will emit the new source value
 		const sourceEvent$ = new Subject<Promise<T> | Observable<T>>();
 
-		// will hold the current value
-		const sourceValue = signal<T | undefined>(
-			options?.initialValue ?? undefined,
-		);
+		// sourceValue is a signal that will hold the current value and the state of the value
+		let sourceValue: WritableSignal<State<T>>;
+
+		if (options?.requireSync && options?.initialValue === undefined) {
+			const initialComputation = computation(undefined);
+
+			// we don't support promises with requireSync and no initialValue
+			if (isPromise(initialComputation)) {
+				throw new Error(REQUIRE_SYNC_PROMISE_MESSAGE);
+			}
+
+			sourceValue = signal<State<T>>({ kind: StateKind.NoValue });
+
+			if (isObservable(initialComputation)) {
+				initialComputation
+					.subscribe({
+						next: (value) => sourceValue.set({ kind: StateKind.Value, value }),
+						error: (error) => sourceValue.set({ kind: StateKind.Error, error }),
+					})
+					// we need to unsubscribe because we don't want to keep the subscription
+					// we only care about the initial value
+					.unsubscribe();
+
+				if (sourceValue()!.kind === StateKind.NoValue)
+					throw new Error(REQUIRE_SYNC_ERROR_MESSAGE);
+			} else {
+				sourceValue.set({
+					kind: StateKind.Value,
+					value: initialComputation as T,
+				});
+			}
+		} else {
+			sourceValue = signal<State<T>>({
+				kind: StateKind.Value,
+				value: options?.initialValue,
+			});
+		}
 
 		const effectRef = effect(
 			() => {
 				// we need to have an untracked() here because we don't want to register the sourceValue as a dependency
 				// otherwise, we would have an infinite loop
-				const currentValue = untracked(() => sourceValue());
+				const currentState = untracked(() => sourceValue());
+				const currentValue =
+					currentState.kind === StateKind.Value
+						? currentState.value
+						: undefined;
 
 				const newSource = computation(currentValue);
 
-				if (!isObservable(newSource) && !isPromise(newSource)) {
-					// if the new source is not an observable or a promise, we set the value immediately
-					untracked(() => sourceValue.set(newSource));
-				} else {
+				if (isObservable(newSource) || isPromise(newSource)) {
 					// we untrack the source$.next() so that we don't register other signals as dependencies
 					untracked(() => sourceEvent$.next(newSource));
+				} else {
+					// if the new source is not an observable or a promise, we set the value immediately
+					untracked(() =>
+						sourceValue.set({ kind: StateKind.Value, value: newSource as T }),
+					);
 				}
 			},
 			{ injector: options?.injector },
```

Comment:
nit: this is not needed because `assertInjector` ensures this `effect()` runs in the provided Injector if it was provided

Reply:
I added it before that, so removing it now.